integrations/event-handler/state.go

```diff
@@ -238,7 +245,91 @@ func (s *State) SetSessionIndex(id string, index int64) error {
 	return s.dv.Write(sessionPrefix+id, b)
 }
 
-// RemoveSession removes session from the state
+// SetMissingRecording writes the session with missing recording into state.
+func (s *State) SetMissingRecording(sess session, attempt int) error {
+	b, err := json.Marshal(missingRecording{
+		Index:     sess.Index,
+		Attempt:   attempt,
+		Timestamp: sess.UploadTime,
+	})
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	if err := s.dv.Write(missingRecordingPrefix+sess.ID, b); err != nil {
+		return trace.Wrap(err)
+	}
+
+	return trace.Wrap(s.RemoveSession(sess.ID))
+}
+
+type missingRecording struct {
+	ID        string `json:"id,omitempty"`
+	Index     int64
+	Attempt   int
+	Timestamp time.Time
+}
+
+// IterateMissingRecordings finds any sessions with a missing recording and
+// provides them to the callback for processing.
+func (s *State) IterateMissingRecordings(callback func(s session, attempts int) error) error {
+	closeC := make(chan struct{})
+	defer close(closeC)
+	for key := range s.dv.KeysPrefix(missingRecordingPrefix, closeC) {
+		b, err := s.dv.Read(key)
+		if err != nil {
+			return trace.Wrap(err)
+		}
```

Comment:
Let's say this takes too long to process and someone removes a file on disk.
If that's the case, I assume we get some kind of `file not found` error. Should we continue the loop in this case (with a warning log)?